src/helpers/fetchEndpointWithCorrelationId.test.js

```diff
@@ -0,0 +1,33 @@
+import fetchEndpointWithCorrelationId from "../helpers/fetchEndpointWithCorrelationId";
+import { v4 as uuidv4 } from "uuid";
+import nock from "nock";
+
+describe("fetchEndpointWithCorrelationId", () => {
+  let body = JSON.stringify({ code: "code" });
+  let correlationId = `${uuidv4()}-login`;
+  let endpoint = "/api/session";
+  let method = "POST";
+  let scope;
+
+  beforeEach(() => {
+    scope = nock("http://localhost:3001", {
+      reqheaders: {
+        "X-Correlation-ID": correlationId,
+        "content-type": "application/json",
+      },
+      allowUnmocked: true,
+    })
+      .post(endpoint, body)
+      .reply(201);
+  });
+  it("can make a request using fetchEndpointWithCorrelationId", async () => {
+    const response = await fetchEndpointWithCorrelationId(
+      method,
+      "http://localhost:3001/api/session",
+      body,
+      correlationId
+    );
+    expect(response.status).toBe(201);
+    scope.isDone();
```

Comment:
do we have a test that shows the backend gets the correlation id?

Reply:
I think this test is implicitly testing that, so as long as this function is used, then the correlationId is being sent as part of the headers. Nock requires the request to be identical, including for all of the headers to be set and will only then respond with a 201. There was a test which tested this assumption but was removed because it was acting up remotely.  All the api tests are use stubs and spies for the request, so they dont really test them at the integration level and as there is nothing that consumes the correlation id, we cant really test it in this way.